dvc/repo/stage.py

```diff
@@ -462,7 +462,8 @@ def is_dvcfile_and_not_ignored(root, file):
 
         def is_out_or_ignored(root, directory):
             dir_path = f"{root}{sep}{directory}"
-            return dir_path in outs or is_ignored(dir_path)
+            # trailing slash needed to check if a directory is gitignored
+            return dir_path in outs or is_ignored(f"{dir_path}/")
```

Comment:
Path sep won't be `/` on windows

```suggestion
            return dir_path in outs or is_ignored(f"{dir_path}{sep}")
```

Reply:
Indeed, I don't know how I forgot about this 

Sorry about the mess